### PR TITLE
Facility upload header validaiton

### PR DESCRIPTION
**Pull request by burhanrashid52**

### CODE REVIEW

**Review by dileepbapat — CHANGES_REQUESTED** [review]

File: backend/portal_api/pkg/db/db.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+var db *gorm.DB
+
+type CSVUploads struct {
+	gorm.Model
+
+	// filename
+	Filename string
+
+	// status
```
Comment (dileepbapat) [inline backend/portal_api/pkg/db/db.go:20]: avoid such comments that doesnt add additional information like //status. Field itself is status

File: backend/portal_api/pkg/handler.go [hunk 2]
```diff
@@ -142,15 +146,119 @@ func postEnrollmentsHandler(params operations.PostEnrollmentsParams, principal *
...
+	uploadEntry.UploadType = "Facility"
+	uploadEntry.TotalRecords = 0
+	uploadEntry.TotalErrorRows = 0
+	db.CreateCSVUpload(&uploadEntry)
+	return uploadEntry
+}
+
+func validateRow(data Scanner) []string {
+	var errorMsgs []string
+	if data.Text("facilityCode") == "" {
```
Comment (dileepbapat) [inline backend/portal_api/pkg/handler.go:206]: Put list of required fields and loop over it and validate.

File: backend/portal_api/pkg/handler.go [hunk 3]
```diff
@@ -142,15 +146,119 @@ func postEnrollmentsHandler(params operations.PostEnrollmentsParams, principal *
...
 	data := NewScanner(params.File)
-	defer params.File.Close()
+	headerErrors := validateHeaders(columns, data)
+	if headerErrors != nil {
+		return operations.NewPostFacilitiesBadRequest().WithPayload(headerErrors)
+	}
+
+	_, fileHeader, _ := params.HTTPRequest.FormFile("file")
+	fileName := fileHeader.Filename
+	preferredUsername := getUserName(params.HTTPRequest)
```
Comment (dileepbapat) [inline backend/portal_api/pkg/handler.go:158]: You dont need get Username as principal is already there in arg.  May be you might have found reference elsewhere when the principal model was not there. We should clean up that too

**Review by dileepbapat — CHANGES_REQUESTED** [review]

File: backend/portal_api/pkg/handler.go (hunk 2 above)
Comment (burhanrashid52) [reply backend/portal_api/pkg/handler.go:206]: Yup. Make more sense. Will do the changes.

File: backend/portal_api/pkg/handler.go [hunk 4]
```diff
@@ -142,12 +144,36 @@ func postEnrollmentsHandler(params operations.PostEnrollmentsParams, principal *
...
 
 func postFacilitiesHandler(params operations.PostFacilitiesParams, principal *models.JWTClaimBody) middleware.Responder {
+	/*	data := NewScanner(params.File)
+		defer params.File.Close()
+		for data.Scan() {
+			createFacility(&data, params.HTTPRequest.Header.Get("Authorization"))
+			log.Info(data.Text("serialNum"), data.Text("facilityName"))
+		}*/
+	columns := strings.Split(config.Config.Facility.Upload.Columns, ",")
+	log.Println(params.File)
```
[the remark this answers is not in the document]
Comment (burhanrashid52) [reply backend/portal_api/pkg/handler.go]: Done

File: backend/portal_api/pkg/handler.go [hunk 5]
```diff
@@ -142,12 +144,36 @@ func postEnrollmentsHandler(params operations.PostEnrollmentsParams, principal *
...
+				Code:    &code,
+				Message: &message,
+			}
+			return operations.NewPostFacilitiesBadRequest().WithPayload(e)
+		}
+	}
+
 	for data.Scan() {
-		createFacility(&data, params.HTTPRequest.Header.Get("Authorization"))
+		_ = createFacility(&data, params.HTTPRequest.Header.Get("Authorization"))
```
[the remark this answers is not in the document]
Comment (burhanrashid52) [reply backend/portal_api/pkg/handler.go]: Done

**Comment by burhanrashid52** [comment]
This PR is a subset of #151 . Fix the review comment in #151 . Hence closing this PR

**Pull request closed without merging**